proctor-common/src/main/java/com/indeed/proctor/common/ProctorResult.java

```diff
@@ -102,6 +105,29 @@ public ProctorResult(
                 (testDefinitions == null) ? emptyMap() : new HashMap<>(testDefinitions));
     }
 
+    @Deprecated
+    public ProctorResult(
+            final String matrixVersion,
+            @Nonnull final Map<String, TestBucket> buckets,
+            @Nonnull final Map<String, Allocation> allocations,
+            // allowing null for historical reasons
+            @Nullable final Map<String, ConsumableTestDefinition> testDefinitions,
+            @Nonnull final Map<String, PayloadProperty> properties) {
+        // Potentially client applications might need to build ProctorResult instances in each
```

Comment:
This is a great comment! Should a summary of this also be part of the `@Deprecated` annotation?

Reply:
Updated documentation to include comment on negative performance